blockchain/v2/scheduler.go

```diff
@@ -8,8 +8,88 @@ import (
 	"time"
 
 	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
 )
 
+// Events
+
```

Comment:
Unused events should be removed. They should be two blocks of events, one for incoming and one for outgoing (ideally all prefixed `sc`)